### DISCUSSION TITLE
Make ToolWindow invisible

### DISCUSSION THREAD

**Discussion by ekrogh** [VsixCommunity/Community.VisualStudio.Toolkit#149] [Q&A]
Hi

How do I make a ToolWindow invisible ?

So far I have found 
`this.Visibility = Visibility.Hidden;`

That hides the ToolWindow content, but the window frame and background is still visible.

—
Eigil

**Comment by reduckted** [comment]
You can use the `FindWindowAsync` method to find your tool window, and then hide it.

```cs
WindowFrame? window = await VS.Windows.FindWindowAsync(typeof(MyToolWindow.Pane).GUID);

if (window is not null)
{
    await window.HideAsync();
}
```

**Reply by reduckted** [reply]
#171 adds a `HideAsync()` method that you can call on the tool window class. For example,

```cs
await RunnerWindow.HideAsync();
```
